### PR TITLE
CLOUDP-79632: Retrieve details for one private endpoint service for AWS or Azure in an Atlas project

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-79632

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli atlas privateEndpoint list
Command "list" is deprecated, Please use mongocli atlas privateEndpoint(s)|privateendpoint(s) aws|azure list|ls [--projectId projectId]
ID                         ENDPOINT SERVICE                                          STATUS             ERROR
5f3fb2c58ef2cb1c795cddbb   com.amazonaws.vpce.us-east-2.vpce-svc-09c771f7ad2a628e8   WAITING_FOR_USER
5f999db19900d15068d425c9   com.amazonaws.vpce.us-east-1.vpce-svc-079752da00a0d26a4   WAITING_FOR_USER
5fff12b1fd7c32762a6d7e30   com.amazonaws.vpce.eu-west-1.vpce-svc-074a4307981953ad4   WAITING_FOR_USER
```

```
./bin/mongocli atlas privateEndpoint describe 5f3fb2c58ef2cb1c795cddbb
Command "describe" is deprecated, Please use mongocli atlas privateEndpoint(s)|privateendpoint(s) aws|azure describe <ID> [--projectId projectId]
ID                         ENDPOINT SERVICE                                          STATUS             ERROR
5f3fb2c58ef2cb1c795cddbb   com.amazonaws.vpce.us-east-2.vpce-svc-09c771f7ad2a628e8   WAITING_FOR_USER

```

```
./bin/mongocli atlas privateEndpoint aws list
ID                         ENDPOINT SERVICE                                          STATUS             ERROR
5f3fb2c58ef2cb1c795cddbb   com.amazonaws.vpce.us-east-2.vpce-svc-09c771f7ad2a628e8   WAITING_FOR_USER
5f999db19900d15068d425c9   com.amazonaws.vpce.us-east-1.vpce-svc-079752da00a0d26a4   WAITING_FOR_USER
5fff12b1fd7c32762a6d7e30   com.amazonaws.vpce.eu-west-1.vpce-svc-074a4307981953ad4   WAITING_FOR_USER
```

```
./bin/mongocli atlas privateEndpoint azure describe 5fff120253b6064ef7de897e
ID                         ENDPOINT SERVICE               STATUS      ERROR
5fff120253b6064ef7de897e   pls_5fff120253b6064ef7de897e   AVAILABLE
```

```
./bin/mongocli atlas privateEndpoint azure list
ID                         ENDPOINT SERVICE               STATUS      ERROR
5fff120253b6064ef7de897e   pls_5fff120253b6064ef7de897e   AVAILABLE
```

### CODE REVIEW

File: internal/cli/atlas/privateendpoints/describe.go [hunk 1]
```diff
@@ -25,14 +25,15 @@ import (
 )
 
 var describeTemplate = `ID	ENDPOINT SERVICE	STATUS	ERROR
-{{.ID}}	{{.EndpointServiceName}}	{{.Status}}	{{.ErrorMessage}}
+{{.ID}}{{if .EndpointServiceName}}	{{.EndpointServiceName}}{{else}}	PrivateLinkServiceName{{end}}	{{.Status}}	{{.ErrorMessage}}
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/describe.go:28]: [nit] you can move the tab outside of the if and have it just once
```suggestion
{{.ID}}	{{if .EndpointServiceName}}{{.EndpointServiceName}}{{else}}PrivateLinkServiceName{{end}}	{{.Status}}	{{.ErrorMessage}}
```

**Review by gssbzn — COMMENTED** [review]
I was going to approve this but looking at the if made me take a look at the API and I thing we are on the same page that there may be a better way

**Comment by andreaangiolillo** [comment]
@gssbzn ready for review

File: internal/cli/atlas/privateendpoints/aws/describe.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	}
+
+	return opts.Print(r)
+}
+
+// mongocli atlas privateEndpoint(s)|privateendpoint(s) aws describe|get <ID> [--projectId projectId]
+func DescribeBuilder() *cobra.Command {
+	opts := new(DescribeOpts)
+	cmd := &cobra.Command{
+		Use:     "describe",
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/aws/describe.go:58]:
```suggestion
		Use:     "describe <id>",
```

File: internal/cli/atlas/privateendpoints/azure/describe.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	}
+
+	return opts.Print(r)
+}
+
+// mongocli atlas privateEndpoint(s)|privateendpoint(s) azure describe|get <ID> [--projectId projectId]
+func DescribeBuilder() *cobra.Command {
+	opts := new(DescribeOpts)
+	cmd := &cobra.Command{
+		Use:     "describe",
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/azure/describe.go:58]:
```suggestion
		Use:     "describe <id>",
```

File: internal/cli/atlas/privateendpoints/azure/descriptions.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+const (
+	privateEndpoints         = "Manage AZURE private endpoints."
+	listPrivateEndpoints     = "List AZURE Private Endpoints for your project."
+	describePrivateEndpoints = "Return a specific AZURE Private Endpoints for your project."
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/azure/descriptions.go:20]:
```suggestion
	privateEndpoints         = "Manage Azure private endpoints."
	listPrivateEndpoints     = "List Azure Private Endpoints for your project."
	describePrivateEndpoints = "Return a specific Azure Private Endpoints for your project."
```

File: internal/cli/atlas/privateendpoints/describe.go [hunk 5]
```diff
@@ -75,5 +75,7 @@ func DescribeBuilder() *cobra.Command {
 	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
 	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
 
+	cmd.Deprecated = "Please use mongocli atlas privateEndpoint(s)|privateendpoint(s) aws|azure describe <ID> [--projectId projectId]"
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/describe.go:78]: 💯 but let's not show the alias here
```suggestion
	cmd.Deprecated = "Please use mongocli atlas privateEndpoints aws describe <ID> [--projectId projectId]"
```
You can also omit azure as the original command was aws only

File: internal/cli/atlas/privateendpoints/list.go [hunk 6]
```diff
@@ -72,13 +71,13 @@ func ListBuilder() *cobra.Command {
...
 
-	cmd.Flags().StringVar(&opts.provider, flag.Provider, "AWS", usage.ProviderPrivateEndpoint)
-
 	cmd.Flags().IntVar(&opts.PageNum, flag.Page, 0, usage.Page)
 	cmd.Flags().IntVar(&opts.ItemsPerPage, flag.Limit, 0, usage.Limit)
 
 	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
 	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
 
+	cmd.Deprecated = "Please use mongocli atlas privateEndpoint(s)|privateendpoint(s) aws|azure list|ls [--projectId projectId]"
```
Comment (gssbzn) [inline internal/cli/atlas/privateendpoints/list.go:80]: same as previous comment let's keep it short

**Review by gssbzn — CHANGES_REQUESTED** [review]
💯  great work here, left a couple of suggestions on the copy but nothing major

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**